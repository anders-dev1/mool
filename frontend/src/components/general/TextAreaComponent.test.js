import {fireEvent, render, screen} from "@testing-library/react";
import TextArea from "./TextAreaComponent";

test('renders correctly', () => {
   const {container} = render(
       <TextArea name={"testname"} placeholder={"this is the placeholder"} value={"initial value"}/>
   );

   expect(container).toMatchSnapshot();
});

test('input changed, when onChange is provided, calls Callback', () =>{
   // Arrange
   const mockCallback = jest.fn();
   const {container} = render(<TextArea onChange={mockCallback}/>);
   const component = container.querySelector('textarea');

   // Act
   fireEvent.change(component, {target: {value: 'hello'}});

   // Assert
   expect(mockCallback).toBeCalled();
});

test('input left after being interacted with, when onBlur is provided, calls Callback', () =>{
   // Arrange
   const mockCallback = jest.fn();
   const {container} = render(<TextArea onBlur={mockCallback}/>);
   const component = container.querySelector('textarea');

   // Act
   fireEvent.focusOut(component);

   // Assert
   expect(mockCallback).toBeCalled();
});