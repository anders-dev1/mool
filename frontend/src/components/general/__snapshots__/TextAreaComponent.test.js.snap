// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="sc-bczRLJ eyqewe"
  >
    <textarea
      name="testname"
      placeholder="this is the placeholder"
    >
      initial value
    </textarea>
  </div>
</div>
`;
